src/components/CheckIn/CheckInDetails.js

```diff
@@ -0,0 +1,167 @@
+import ItemsList from './ItemsList';
+import CheckInDetailsFooter from './CheckInDetailsFooter';
+import checkInItems from './util';
+
+class CheckInDetails extends Component {
+  static manifest = Object.freeze({
+    checkIn: CHECKIN,
+    items: ITEMS,
+    locations: LOCATIONS,
+    query: {},
+  })
+
+  static propTypes = {
+    close: PropTypes.func.isRequired,
+    location: PropTypes.object,
+    mutator: PropTypes.object,
+    pieces: PropTypes.arrayOf(PropTypes.object).isRequired,
+    resources: PropTypes.object,
+  }
+
+  constructor(props) {
+    super(props);
+    this.callout = React.createRef();
+
+    this.state = {
+      isAllChecked: true,
+      isLoading: true,
+      items: [],
+    };
+  }
+
+  componentDidMount() {
+    const { mutator, pieces } = this.props;
+
+    fetchItems(mutator, pieces).then(itemsMap => this.setState(() => {
+      const items = cloneDeep(pieces);
+
+      const updatedItems = items.map(item => {
+        item.barcode = get(itemsMap, [item.itemId, 'barcode'], '');
+        item.itemStatus = ITEM_STATUS.received;
+
+        return item;
+      });
+
+      return {
+        isLoading: false,
+        items: updatedItems,
+      };
+    }));
+  }
+
+  toggleItem = (item) => {
+    this.setState(state => {
+      const items = cloneDeep(state.items);
+      const toggledItem = find(items, { 'id': item.id });
+
+      toggledItem.isChecked = !item.isChecked;
+
+      return {
+        isAllChecked: false,
+        items,
+      };
+    });
+  }
+
+  toggleAll = () => {
+    this.setState((state) => {
+      const isAllChecked = !state.isAllChecked;
+      const items = state.items.map(item => ({
+        ...item,
+        isChecked: isAllChecked,
+      }));
+
+      return {
+        isAllChecked,
+        items,
+      };
+    });
+  }
+
+  submitCheckIn = () => {
+    const { close, location, mutator } = this.props;
+    const checkInItemsList = this.state.items.filter(item => item.isChecked === true);
+
+    checkInItems(checkInItemsList, mutator.checkIn)
+      .then(() => this.callout.current.sendCallout({
+        type: 'success',
+        message: <FormattedMessage id="ui-orders.checkIn.checkInItem.success" />,
+      }))
+      .then(() => close())
+      .then(() => mutator.query.update({
+        _path: location.pathname.replace(CHECKIN_URLS.items, CHECKIN_URLS.history),
+      }))
+      .catch(() => this.callout.current.sendCallout({
+        type: 'error',
+        message: <FormattedMessage id="ui-orders.checkIn.checkInItem.error" />,
+      }));
+  }
+
+  onChangeField = (item, value, key) => {
+    this.setState((state) => {
+      const items = cloneDeep(state.items);
+      const selectedItem = find(items, { 'id': item.id });
+
+      selectedItem[key] = value;
+
+      return { items };
+    });
+  }
+
+  render() {
+    const { close, resources } = this.props;
+    const { isAllChecked, items, isLoading } = this.state;
+
+    return (
+      <Modal
```

Comment:
IMO, everything should be moved to the presentation, not only `ItemsList`

Reply:
Ok